x/oracle/module.go

```diff
@@ -203,155 +203,160 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 			am.keeper.RemoveNonceWithFeederIDForValidators(ctx, feederID, agc.GetValidators())
 		}
 	}()
-	// update&check slashing info
-	validatorPowers := agc.GetValidatorPowers()
-	validators := make([]string, 0, len(validatorPowers))
-	for validator := range validatorPowers {
-		validators = append(validators, validator)
-	}
-	sort.Strings(validators)
-	for _, validator := range validators {
-		power := validatorPowers[validator]
-		reportedInfo, found := am.keeper.GetValidatorReportInfo(ctx, validator)
-		if !found {
-			logger.Error(fmt.Sprintf("Expected report info for validator %s but not found", validator))
-			continue
+	latestValidatorUpdateBlock, ok := am.keeper.GetValidatorUpdateBlock(ctx)
+	p := am.keeper.GetParams(ctx)
+
+	if (!ok || (latestValidatorUpdateBlock.Block <= uint64(ctx.BlockHeight())-uint64(p.MaxNonce))) && len(validatorUpdates) == 0 {
```

Comment:
So slashing can only happen if either
1. there is no last validator update block, or
2. the last update was within the nonce size and there have been no updates in this block.
is that correct?